core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -951,6 +970,10 @@ public Thread newThread(Runnable r) {
             return threadPool;
         }
 
+        public Double getIdlePercent() {
+            return 1 - (double) scheduledThreadPool.getActiveCount() / (double) scheduledThreadPool.getCorePoolSize();
```

Comment:
Please help me understand why are we using getCorePoolSize() in stead of getPoolSize() here?

Reply:
A ScheduledThreadPoolExecutor takes the `corePoolSize` in its constructor, and it creates a fixed-size pool using the corePoolSize.
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledThreadPoolExecutor.html#ScheduledThreadPoolExecutor-int-

> While this class inherits from [ThreadPoolExecutor](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html), a few of the inherited tuning methods are not useful for it. In particular, because it acts as a fixed-sized pool using corePoolSize threads

Since corePoolSize controls the pool size for a ScheduledThreadPoolExecutor, it seems better to use it here.

Reply:
Thank you for the explanation.